documenters_aggregator/spiders/ilrb.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+"""
+All spiders should yield data shaped according to the Open Civic Data
+specification (http://docs.opencivicdata.org/en/latest/data/event.html).
+"""
+import scrapy
+
+import re
+from datetime import datetime, timedelta
+from pytz import timezone
+
+
+class IlrbSpider(scrapy.Spider):
+    name = 'ilrb'
+    allowed_domains = ['https://www.illinois.gov']
+    start_urls = ['https://www.illinois.gov/ilrb/meetings/Pages/default.aspx']
+
+    """
+    This page only lists the next upcoming meeting for each of the three boards.
+    All other meetingd dates are `proposed` and only available via PDF.
+    """
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` a dict that follows the `Open Civic Data
+        event standard <http://docs.opencivicdata.org/en/latest/data/event.html>`_.
+
+        Change the `_parse_id`, `_parse_name`, etc methods to fit your scraping
+        needs.
+        """
+
+        # There's not a lot of structure on this page, so this selector is pretty fragile
+        for item in response.css('.soi-article-content .row-fluid .span12>p:nth-child(odd)'):
+            """
+            Some monthly meetings are skipped. Instead of providing a date,
+            there's text that says 'No /name/ meeting in month'.
+            If the date/time info can't be parsed, assume that it is a `no meeting`
+            notice.
+            """
+            start_datetime = self._parse_start(item)
+            if start_datetime is None:
+                continue
+            name = self._parse_name(item)
+            yield {
+                '_type': 'event',
+                'id': self._generate_id(self._format_date(start_datetime), name),
+                'name': name,
+                'description': self._parse_description(item),
+                'classification': self._parse_classification(item),
+                'start_time': self._format_date(start_datetime),
+                'end_time': self._calc_end(start_datetime, hours=1),
+                'all_day': self._parse_all_day(item),
+                'status': self._parse_status(item),
+                'location': self._parse_location(item),
+            }
+
+    def _generate_id(self, start_time, name):
+        """
+        We use the start time to generate an ID since there is no publically
+        exposed meeting ID.
+        """
+
+        date = start_time.split('T')[0]
+        dashified = re.sub(r'[^a-z]+', '-', name.lower())
```

Comment:
Not a blocker, but it's possible that [python-slugify](https://github.com/un33k/python-slugify) is a little safer option.